### ISSUE TITLE
Model optimization for rhythmic time courses 

### ISSUE THREAD

**Issue by lindseypower** [jonescompneurolab/hnn-core#176]
Is there functionality in HNN or HNN-core for optimizing parameters to match a rhythmic input (e.g. a transient beta time course) or is the model optimization only available for ERP data?

**Comment by blakecaldwell** [comment]
This likely be part of `hnn-core.analysis` #139 

Implementing optimization in `hnn-core` is still missing. The relevant code is in #77

**Comment by jasmainak** [comment]
#77 sounds like a future hackathon project :)

**Comment by rythorpe** [comment]
No, we'll need a different optimization design. The idea behind the current sequential design is that it tunes only those parameters that are relevant for a given time window while keeping the others fixed, thus drastically reducing the dimensionality of the optimization problem (e.g., an early evoked drive can be tuned for the early portion of an evoked response simulation while a late evoked drive can be tuned for the later portion of the same simulation). For stationary signals like baseline noise and oscillations (corresponding to Poisson drive and rhythmic bursty drive), the parameter space for drives can't be segmented into different time windows, per the definition of stationarity.

**Comment by ntolley** [comment]
Closed now that #673 is merged

I'll note that it handles a rather specific optimization objective (band power in user defined frequency ranges), but this covers a wide range of the questions/problems people are concerned with when using rhythmic inputs
